Sources/MapboxSpeech/MBSpeechOptions.swift

```diff
@@ -85,3 +85,30 @@ open class SpeechOptions: Codable {
         return params
     }
 }
+
+public extension Locale {
+    
+    /**
+     `String` Returns the identifier of the locale identifier supported by Amazon Polly [`Supported Language`](https://docs.aws.amazon.com/polly/latest/dg/SupportedLanguage.html).
+     
+     While common language identifiers are two-letter `ISO 639-1` standard, three-letter `ISO 639-2` standard or even `RFC 4647` (known as BCP 47), `Amazon Polly` uses `ISO 639-3`
+     W3C language identification which creates incompatibility with `Locale.current`.
+     This computed property either return `Locale.identifier` or the supported version of the unknown code.
+     List of currently unsupported codes :
+     "ar-SA", "zh-CN", "zh-HK", "zh-Hans", "zh-Hant", "zh-TW"
+     */
+    var amazonIdentifier : String {
+        let unsupported : Dictionary<String, String> = [
+            "ar"     : "arb",
+            "zh"     : "cmn-CN"
+        ]
+        let languageComponents = Locale.components(fromIdentifier: self.identifier)
+        if let languageCode = languageComponents["kCFLocaleLanguageCodeKey"] {
+            if unsupported.keys.contains(languageCode), let patchedIdentifier = unsupported[ languageCode ] {
```

Comment:
`Locale.languageCode` is a more straightforward way to get the ISO&nbsp;639 code from the `Locale`. Or does `kCFLocaleLanguageCodeKey` account for any edge cases that `Locale.languageCode` would not account for?

Reply:
`kCFLocaleLanguageCodeKey` does actually return the needed ISO 639-x/IETF BCP 47 while `Locale.languageCode` tend to return Locale language + Country Code. ex: for an iPhone's default language as English and setup in France, will return `en_FR`. I will add the key differences in the documentation.

Reply:
Interesting, this is something we might want to account for in mapbox-directions-swift too, for consistency. 🤔